src/BEditaClient.php

```diff
@@ -606,6 +606,17 @@ protected function sendRequest(string $method, string $path, ?array $query = nul
      */
     protected function requestUri(string $path, ?array $query = null) : Uri
     {
+        // if path contains query strings, remove them from path and add them to query filter
+        $url = parse_url($path);
```

Comment:
I would move this pieace of logic on the bottom where treat `$query` with something like

```
parse_str($uri->getQuery(), $uriQuery);
if ($query) {
    $query = array_merge((array)$uriQuery, (array)$query);
    $uri = $uri->withQuery(http_build_query($query));
}
```